internal/services/sources_service.go

```diff
@@ -20,10 +20,17 @@ func ListSources(w http.ResponseWriter, r *http.Request) {
 	}
 	resp, err := client.ListApplicationTypeSourcesWithResponse(r.Context(), config.Sources.AppTypeId, &sources.ListApplicationTypeSourcesParams{}, AddIdentityHeader)
 	if err != nil {
-		renderError(w, r, payloads.New3rdPartyClientError(r.Context(), "list sources", err))
+		renderError(w, r, payloads.SourcesClientError(r.Context(), "list sources", err, resp.StatusCode()))
+		return
+	}
+	if resp.StatusCode() == 404 {
```

Comment:
This could be other helper function like `IsHTTPNotFound` or something like that.